app/models/user.rb

```diff
@@ -316,9 +316,15 @@ def liked_pages
   end
 
   def send_digest_email
-    top_picks = content_followed_in_period(1.week.ago, Time.now)
-    if top_picks.count.positive?
-      SubscriptionMailer.send_digest(id, top_picks).deliver_now
+    if self.has_tag('digest:daily')
+      top_picks = content_followed_in_period(Time.now - 1.day, Time.now)
+      k = 0
+    elsif self.has_tag('digest:weekly')
+      top_picks = content_followed_in_period(Time.now - 1.week, Time.now)
+      k = 1
+    end
+    if top_picks.count > 0
```

Comment:
This is where the error points to. 

```

ERROR["test_send_digest_email", #<Minitest::Reporters::Suite:0x00005572aa9d18a0 @name="UserTest">, 24.959422129000018]
 test_send_digest_email#UserTest (24.96s)
NoMethodError:         NoMethodError: undefined method `count' for nil:NilClass
        Did you mean?  concat
            app/models/user.rb:326:in `send_digest_email'
            test/unit/user_test.rb:225:in `block in <class:UserTest>'
```

Reply:
Oh yes, so probably because the daily top_picks were empty, that's why it couldn't trigger the condition, should I put a nil check there?

Reply:
So I tried to a check like `!top_picks.nil? && top_picks.count > 0` ,  however another error showed up in travis.

Reply:
Sure or as i said above, you can start by defining it as `[]`. Great!

Reply:
I have tried both, but still, the new error prevails, could you please tell me why?